### PR TITLE
[DROOLS-5658] [DROOLS-4844] [DROOLS-5657] [DROOLS-2836] [DROOLS-5699] 

### PR DESCRIPTION
**Pull request by lucamolteni**
**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

https://issues.redhat.com/browse/DROOLS-5658
https://issues.redhat.com/browse/DROOLS-2836
https://issues.redhat.com/browse/DROOLS-5657
https://issues.redhat.com/browse/DROOLS-4844

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1498
https://github.com/kiegroup/drools/pull/3159
https://github.com/kiegroup/droolsjbpm-knowledge/pull/471
https://github.com/kiegroup/droolsjbpm-integration/pull/2277

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by lucamolteni** [comment]
jenkins test again

**Comment by lucamolteni** [comment]
jenkins test again

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/GenerateANCMojo.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    private Map<String, String> properties;
+
+    @Parameter(required = true, defaultValue = "${project}")
+    private MavenProject project;
+
+    @Parameter(required = true, defaultValue = "${project.build.outputDirectory}")
+    private File outputDirectory;
+
+    @Parameter(property = "generateModel", defaultValue = "YES_WITHDRL")
+    private String generateModel;
```
Comment (tarilabs) [inline kie-maven-plugin/src/main/java/org/kie/maven/plugin/GenerateANCMojo.java:82]: you might want to "move up" similarly to here: https://github.com/kiegroup/droolsjbpm-integration/blob/32f44c724770dfd3eca136d3cacdcf0b90455a1e/kie-maven-plugin/src/main/java/org/kie/maven/plugin/AbstractDMNValidationAwareMojo.java#L63-L68

File: kie-plugins-testing/src/test/projects/kjar-9-declared-types/pom.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
   </properties>
-
```
Comment (tarilabs) [inline kie-plugins-testing/src/test/projects/kjar-9-declared-types/pom.xml:32]: this file modification is not necessary

File: kie-plugins-testing/src/test/resources/logback-test.xml [hunk 3]
```diff
@@ -12,6 +12,7 @@
 
   <logger name="org.drools.compiler.xpath.tobeinstrumented" level="INFO"/>
   <logger name="org.kie" level="INFO"/>
+  <logger name="org.drools.ancompiler" level="debug"/>
```
Comment (tarilabs) [inline kie-plugins-testing/src/test/resources/logback-test.xml:15]: are you sure you want to leave this as `debug`?
```suggestion
  <logger name="org.drools.ancompiler" level="INFO"/>
```

**Review by tarilabs — APPROVED** [review]
LGTM with some minor comments

How did you solved btw that kie-ci from the plugin is attempting to resolve Drools libraries at `1.0.0.Final` version btw?

**Comment by lucamolteni** [comment]
jenkins test again

File: kie-maven-plugin/src/main/java/org/kie/maven/plugin/AbstractDMNValidationAwareMojo.java [hunk 4]
```diff
@@ -63,10 +65,21 @@
     @Parameter(property = "validateDMN", defaultValue = "VALIDATE_SCHEMA,VALIDATE_MODEL,ANALYZE_DECISION_TABLE")
     private String validateDMN;
 
+    @Parameter(property = "generateModel", defaultValue = "YES_WITHDRL") // DROOLS-5663 align kie-maven-plugin default value for generateModel configuration flag
+    private String generateModel;
+
```
Comment (tarilabs) [inline kie-maven-plugin/src/main/java/org/kie/maven/plugin/AbstractDMNValidationAwareMojo.java:70]: please 1 move up more. This class is called `...DMNValidationAware...`
Kindly consider move this stuff up to `AbstractKieMojo`

**Comment by lucamolteni** [comment]
> How did you solved btw that kie-ci from the plugin is attempting to resolve Drools libraries at `1.0.0.Final` version btw?

https://issues.redhat.com/browse/DROOLS-5725

**Review by tkobayas — APPROVED** [review]

File: kie-plugins-testing/src/test/resources/logback-test.xml (hunk 3 above)
Comment (lucamolteni) [reply kie-plugins-testing/src/test/resources/logback-test.xml:15]: Yes it doesn't generate much output

**Review by mariofusco — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]

File: kie-maven-plugin/src/main/resources/META-INF/plexus/components.xml [hunk 5]
```diff
@@ -9,7 +9,16 @@
...
-                    <compile>org.apache.maven.plugins:maven-compiler-plugin:compile,org.kie:kie-maven-plugin:generateModel,org.kie:kie-maven-plugin:generateDMNModel,org.kie:kie-maven-plugin:generatePMMLModel,org.apache.maven.plugins:maven-compiler-plugin:compile,org.kie:kie-maven-plugin:build</compile>
+                    <compile>
+                        org.apache.maven.plugins:maven-compiler-plugin:compile,
+                        org.kie:kie-maven-plugin:generateModel,
+                        org.kie:kie-maven-plugin:generateDMNModel,
+                        org.kie:kie-maven-plugin:generatePMMLModel,
+                        org.apache.maven.plugins:maven-compiler-plugin:compile,
+                        org.kie:kie-maven-plugin:build,
+                        org.kie:kie-maven-plugin:generateANC,
+                        org.apache.maven.plugins:maven-compiler-plugin:compile
```
Comment (danielezonca) [inline kie-maven-plugin/src/main/resources/META-INF/plexus/components.xml:20]: Are you sure `build` is not the last one?
Btw I think we should review this mechanism, now we have three maven:compile steps :thinking:

Comment (lucamolteni) [reply kie-maven-plugin/src/main/resources/META-INF/plexus/components.xml:20]: I know, but unfortunately ANC compiler needs the rete and the rete is only created at runtime. Then it has to generate those classes and compile them.

**Comment by lucamolteni** [comment]
jenkins test again

**Comment by lucamolteni** [comment]
jenkins test again

**Comment by lucamolteni** [comment]
Jenkins retest this

**Review by danielezonca — APPROVED** [review]

**Comment by lucamolteni** [comment]
github 8 action is failing for 2020-10-15T11:12:29.9033429Z ##[error]No space left on device

**Comment by lucamolteni** [comment]
Failing tests are unrelated
```
org.kie.server.integrationtests.jbpm.QueryDataServiceIntegrationTest.testErrorHandlingFailedToSignal[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.QueryDataServiceIntegrationTest.testErrorHandlingFailedToSignal[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.QueryDataServiceIntegrationTest.testErrorHandlingFailedToSignal[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.QueryDataServiceIntegrationTest.testErrorHandlingFailedToSignal[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.QueryDataServiceIntegrationTest.testErrorHandlingFailedToSignal[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.QueryDataServiceIntegrationTest.testErrorHandlingFailedToSignal[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.RuntimeDataServiceIntegrationTest.testQueryProcessesByVariables[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.RuntimeDataServiceIntegrationTest.testQueryProcessesByVariables[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.UserTaskServiceIntegrationTest.testFindTasksByVariable[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.UserTaskServiceIntegrationTest.testFindTasksByVariable[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.UserTaskServiceIntegrationTest.testFindTasksByVariable[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:37498/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.UserTaskServiceIntegrationTest.testFindTasksByVariable[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.UserTaskServiceIntegrationTest.testFindTasksByVariable[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.UserTaskServiceIntegrationTest.testFindTasksByVariable[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]
```

**Comment by lucamolteni** [comment]
jenkins test this

**Pull request merged**